src/ctelnet.cpp

```diff
@@ -630,38 +809,219 @@ void cTelnet::slot_socketSslError(const QList<QSslError>& errors)
     }
 
     if (mpHost->mSslIgnoreAll) {
-        mpSocket.ignoreSslErrors(errors);
+        pSocket->ignoreSslErrors(errors);
     } else {
-        mpSocket.ignoreSslErrors(ignoreErrorList);
+        pSocket->ignoreSslErrors(ignoreErrorList);
     }
 }
 #endif
 
+// We now need to examing hostInfo to see whether it has IPv4 or IPv6 or both
+// types of address - so we know which socket (or both) to try to connect.
 void cTelnet::slot_socketHostFound(QHostInfo hostInfo)
 {
+#if defined(DEBUG_TELNET) && (DEBUG_TELNET & 4)
+    qDebug().noquote() << "cTelnet::slot_socketHostFound(QHostInfo) INFO - called.";
+#endif
+    QStringList addressList_ipV4;
+    QStringList addressList_ipV6;
+    for (const QHostAddress& address : hostInfo.addresses()) {
+        // Handle (or not) some special cases:
+        if (QHostAddress::Null == address
+            || QHostAddress::Broadcast == address
+            || QHostAddress::Any == address
+            || QHostAddress::AnyIPv4 == address
+            || QHostAddress::AnyIPv6 == address) {
+
+            continue;
+        }
+
+        if (QHostAddress::LocalHostIPv6 == address) {
+            addressList_ipV6.append(qsl("::1"));
+            continue;
+        }
+
+        if (QHostAddress::LocalHost == address) {
+            addressList_ipV4.append(qsl("127.0.0.1"));
+            continue;
+        }
+
+        switch (address.protocol()) {
+        case QAbstractSocket::IPv4Protocol:
+            addressList_ipV4.append(address.toString());
+            break;
+        case QAbstractSocket::IPv6Protocol:
+            addressList_ipV6.append(address.toString());
+            break;
+        case QAbstractSocket::AnyIPProtocol:
+            [[fallthrough]];
+        case QAbstractSocket::UnknownNetworkLayerProtocol:
+            {} // No-op
+        }
+    }
+
+    const bool hasIPv4_address = (addressList_ipV4.count());
+    const bool hasIPv6_address = (addressList_ipV6.count());
+    if (!(hasIPv4_address || hasIPv6_address)) {
+        // Haven't got any address to use:
+        postMessage(tr("[ ERROR ] - Host name lookup Failure! A connection cannot be established.\n"
+                                   "The server name is not correct, or your nameservers are not\n"
+                                   "working properly."));
+        return;
+    }
+
+    // Report found IP addresses:
+    QStringList addressesToReport;
+    for (const auto& address : addressList_ipV6) {
+        /*: Used to add an IPv6 address line to the list displayed during
+         * connecting to a Host.
+         */
+        addressesToReport << tr("%1 (IPv6)").arg(address);
```

Comment:
What do you want translators to translate here?
Isn't IPv6 an established technical term?
Not sure if this can be qsl() instead.